contracts/vesting-lp/src/contract.rs

```diff
@@ -36,11 +50,404 @@ pub fn execute(
+        .querier
+        .query_balance(env.contract.address.to_string(), paired_asset_denom.clone())?
+        .amount;
+
+    let mut msgs: Vec<CosmosMsg> = vec![];
+
+    // push message to withdraw liquidity from the xyk pair
+    if !amount.is_zero() {
+        msgs.push(CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: xyk_lp_token.to_string(),
+            msg: to_binary(&Cw20ExecuteMsg::Send {
+                contract: xyk_pair.to_string(),
+                amount,
+                msg: to_binary(&PairCw20HookMsg::WithdrawLiquidity { assets: vec![] })?,
+            })?,
+            funds: vec![],
+        }))
+    }
+    let config = CONFIG.load(deps.storage)?;
+    vesting_state(config.extensions.historical).update::<_, ContractError>(
+        deps.storage,
+        env.block.height,
+        |s| {
+            let mut state = s.unwrap_or_default();
+            state.total_released = state.total_released.checked_add(amount)?;
+            Ok(state)
+        },
+    )?;
+    // push the next migration step as a callback message
+    msgs.push(
+        CallbackMsg::ProvideLiquidityToClPairAfterWithdrawal {
+            ntrn_denom,
+            ntrn_init_balance,
+            paired_asset_denom,
+            paired_asset_init_balance,
+            cl_pair,
+            slippage_tolerance,
+            user,
+        }
+        .to_cosmos_msg(&env)?,
+    );
+
+    Ok(Response::default().add_messages(msgs))
+}
+
+#[allow(clippy::too_many_arguments)]
+fn provide_liquidity_to_cl_pair_after_withdrawal_callback(
+    deps: DepsMut,
+    env: Env,
+    ntrn_denom: String,
+    ntrn_init_balance: Uint128,
+    paired_asset_denom: String,
+    paired_asset_init_balance: Uint128,
+    cl_pair_address: Addr,
+    slippage_tolerance: Decimal,
+    user: VestingAccountResponse,
+) -> Result<Response, ContractError> {
+    let ntrn_balance_after_withdrawal = deps
+        .querier
+        .query_balance(env.contract.address.to_string(), ntrn_denom.clone())?
+        .amount;
+    let paired_asset_balance_after_withdrawal = deps
+        .querier
+        .query_balance(env.contract.address.to_string(), paired_asset_denom.clone())?
+        .amount;
+
+    // calc amount of assets that's been withdrawn
+    let withdrawn_ntrn_amount = ntrn_balance_after_withdrawal.checked_sub(ntrn_init_balance)?;
+    let withdrawn_paired_asset_amount =
+        paired_asset_balance_after_withdrawal.checked_sub(paired_asset_init_balance)?;
+
+    let mut msgs: Vec<CosmosMsg> = vec![];
+
+    let migration_config: XykToClMigrationConfig = XYK_TO_CL_MIGRATION_CONFIG.load(deps.storage)?;
+
+    let balance_response: BalanceResponse = deps.querier.query_wasm_smart(
+        migration_config.new_lp_token,
+        &Cw20QueryMsg::Balance {
+            address: env.contract.address.to_string(),
+        },
+    )?;
+    let current_balance = balance_response.balance;
+
+    if !withdrawn_ntrn_amount.is_zero() && !withdrawn_paired_asset_amount.is_zero() {
```

Comment:
I. just excessive check for psycological safety. potentially of course they could not be 0
II.  > which means that we used a wrong dust param value
 this would bee unreachable then, because `zeroAmout` err would be thrown on withdraw.

Reply:
Let's remove them:
* it's easier to read the code;
* the situation is almost impossible;